File: example/src/main/java/com/example/picassocommons/TargetActivity.java
package com.example.picassocommons;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;

import com.hendraanggrian.picasso.commons.target.Targets;
import com.hendraanggrian.support.utils.view.InputMethods;
import com.hendraanggrian.support.utils.widget.Toasts;
import com.squareup.picasso.Picasso;

import butterknife.BindView;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public final class TargetActivity extends BaseActivity {

    @BindView(R.id.toolbar_target) Toolbar toolbar;
    @BindView(R.id.edittext_target) EditText editText;
    @BindView(R.id.button_target_clear) Button buttonClear;
    @BindView(R.id.button_target_go) Button buttonGo;
    @BindView(R.id.imageview_target) ImageView imageView;

    @Override
    protected int getContentView() {
        return R.layout.activity_target;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setSupportActionBar(toolbar);
        buttonClear.setOnClickListener(v -> editText.setText(""));
        buttonGo.setOnClickListener(v -> {
            InputMethods.hideSoftInput(this);
            Picasso.with(this)
                    .load(editText.getText().toString())
                    .into(Targets.placeholder(imageView).callback(
                            (bitmap, from) -> Toasts.showShort(this, "Loaded."),
                            errorDrawable -> Toasts.showShort(this, "Failed."),
                            placeHolderDrawable -> Toasts.showShort(this, "Prepare...")));
        });
    }
}

File: example/src/com/example/picassocommons/TargetActivity.kt
package com.example.picassocommons

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import com.hendraanggrian.kota.view.hideInput
import com.hendraanggrian.picasso.commons.target.Targets
import com.squareup.picasso.getPicasso
import kotlinx.android.synthetic.main.activity_target.*
import org.jetbrains.anko.toast

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class TargetActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setSupportActionBar(toolbar)
        setContentView(R.layout.activity_target)
        buttonClear.setOnClickListener { editText!!.setText("") }
        buttonGo.setOnClickListener {
            hideInput()
            getPicasso().load(editText.text.toString())
                    .into(Targets.placeholder(imageView)
                            .callback({ _, _ ->
                                toast("Loaded.")
                            }, {
                                toast("Failed.")
                            }, {
                                toast("Prepare...")
                            }))
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }
}